internal/manifests/collector/servicemonitor.go

```diff
@@ -111,3 +113,10 @@ func endpointsFromConfig(logger logr.Logger, otelcol v1beta1.OpenTelemetryCollec
 	}
 	return endpoints
 }
+
+// addExtraLabels adds extra labels to the ServiceMonitor labels map.
+func addExtraLabels(labels *map[string]string, extraLabels map[string]string) {
+	for k, v := range extraLabels {
```

Comment:
Maybe we should log something if the label already exists. Because this will overwrite the value.